core/src/server/access_control/matcher.rs

```diff
@@ -32,7 +32,7 @@ use tracing::warn;
 /// Pattern that can be matched to string.
 pub(crate) trait Pattern {
 	/// Returns true if given string matches the pattern.
-	fn matches<T: AsRef<str>>(&self, other: T) -> bool;
+	fn matches<'a, T: AsRef<str> + PartialEq<&'a str>>(&self, other: T) -> bool;
```

Comment:
I don't follow why this additional lifetime and PartialEq trait bound was added?